src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -179,6 +193,26 @@ protected Object invokeStep(StepDescriptor d, Object args) {
             d.checkContextAvailability(context);
             Thread.currentThread().setContextClassLoader(CpsVmExecutorService.ORIGINAL_CONTEXT_CLASS_LOADER.get());
             s = d.newInstance(ps.namedArgs);
+            try {
+                // No point storing empty arguments, and ParallelStep is a special case where we can't store its closure arguments
+                if (ps.namedArgs != null && !(ps.namedArgs.isEmpty()) && isKeepStepInfo() && !(s instanceof ParallelStep)) {
+                    Map<String, Object> actualArgs = ps.namedArgs;
+                    if (d.isMetaStep() && ps.namedArgs.get("delegate") instanceof Map) {
+                        actualArgs = (Map<String,Object>)ps.namedArgs.get("delegate");
+                    }
+                    // Get the environment variables to find ones that might be credentials bindings
+                    Computer comp = context.get(Computer.class);
+                    EnvVars allEnv = new EnvVars(context.get(EnvVars.class));
+                    if (comp != null && allEnv != null) {
+                        allEnv.entrySet().removeAll(comp.getEnvironment().entrySet());
```

Comment:
I do not see how that is orthogonal. This PR is adding considerable technical debt, introducing a lot of complex code whose correctness is far from clear, apparently to avoid making a very simple change in `credentials-binding`.